libgalois/include/katana/TypedPropertyGraph.h

```diff
@@ -0,0 +1,242 @@
+#ifndef KATANA_LIBGALOIS_KATANA_TYPEDPROPERTYGRAPH_H_
+#define KATANA_LIBGALOIS_KATANA_TYPEDPROPERTYGRAPH_H_
+
+#include <tuple>
+
+#include <arrow/type_fwd.h>
+#include <boost/iterator/counting_iterator.hpp>
+
+#include "katana/Details.h"
+#include "katana/NoDerefIterator.h"
+#include "katana/Properties.h"
+#include "katana/PropertyGraph.h"
+#include "katana/PropertyViews.h"
+#include "katana/Result.h"
+#include "katana/Traits.h"
+
+namespace katana {
+
+/// A property graph is a graph that has properties associated with its nodes
+/// and edges. A property has a name and value. Its value may be a primitive
+/// type, a list of values or a composition of properties.
+///
+/// A TypedPropertyGraph is a representation of a property graph that imposes a
+/// typed view on top of an underlying \ref PropertyGraph. A
+/// PropertyGraph is appropriate for cases where the graph is largely
+/// uninterpreted and can be manipulated as a collection of bits. A
+/// TypedPropertyGraph is appropriate for cases where computation needs to be done
+/// on the properties themselves.
+///
+/// \tparam NodeProps A tuple of property types (\ref Properties.h) for nodes
+/// \tparam EdgeProps A tuple of property types for edges
+template <typename NodeProps, typename EdgeProps>
+class TypedPropertyGraph {
+  using NodeView = PropertyViewTuple<NodeProps>;
+  using EdgeView = PropertyViewTuple<EdgeProps>;
```

Comment:
Oh, TypedPropertyGraph, ya mae be foolin' the others with yer fancy name and yer new closes, but I recognize those template arguments as the mark of the Devil! (actually, I thought this was DistGraph, but it is the old PropertyGraph, which I never used)

Reply:
PropertyGraph => TypedPropertyGraph is mainly used in analytics applications themselves, and by the great spirit of Alan Turing, shouldn't find much use within the internal infrastructure of katana itself.